x/collection/keeper/supply.go

```diff
@@ -26,15 +26,12 @@ func (k Keeper) CreateContract(ctx sdk.Context, creator sdk.AccAddress, contract
 		Grantee:    creator.String(),
 	}
 	ctx.EventManager().EmitEvent(collection.NewEventGrantPermTokenHead(eventGrant))
-	for _, permission := range collection.Permission_value {
-		p := collection.Permission(permission)
-		if p == collection.PermissionUnspecified {
-			continue
-		}
+	for i := 1; i < len(collection.Permission_value); i++ {
+		p := collection.Permission(i)
 
 		eventGrant.Permission = p
 		ctx.EventManager().EmitEvent(collection.NewEventGrantPermTokenBody(eventGrant))
-		k.Grant(ctx, contractID, []byte{}, creator, collection.Permission(permission))
+		k.Grant(ctx, contractID, []byte{}, creator, p)
```

Comment:
Is this because `collection.Permission_value` is map type?
If then, how can we guarantee that this modification is deterministic?

Reply:
`for i := 1; i < len(collection.Permission_value); i++ {`
guarantees it.

Reply:
Could you add checking deterministic unittest?

Reply:
We cannot simply check the determinism by checking the apphash after the execution, because, it may not have different root hash (it's the reason why we could not catch this code until now). Any opinions?

Reply:
It might be possible checking the relevant events. But in this case, we should ensure event determinism for the simple unit test code. But the changes would be large, so I cannot simply add the expected modifications in this fix PR.

Reply:
I thought it would be all right if we had a unittest that would put in a list of multiple permissions ahead of time and make sure that these values always write in the same sequence of events.